polaris/competition/_base.py

```diff
@@ -0,0 +1,229 @@
+from datetime import datetime
+import json
+from typing import Callable, TypeAlias, Union
+
+from pydantic import (
+    Field,
+    computed_field,
+    field_serializer,
+    field_validator,
+)
+
+from polaris.dataset._subset import Subset
+from polaris.evaluate import Metric
+from polaris.evaluate._predictions import CompetitionPredictions
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.dict2html import dict2html
+from polaris.utils.types import TargetType, TaskType, SplitType
+
+ColumnsType: TypeAlias = str | list[str]
+
+
+class CompetitionSpecification(DatasetV2):
+    """An instance of this class represents a Polaris competition. It defines fields and functionality
+    that in combination with the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class, allow
+    users to participate in competitions hosted on Polaris Hub.
+
+    Examples:
+        Basic API usage:
+        ```python
+        import polaris as po
+
+        # Load the benchmark from the Hub
+        competition = po.load_competition("polaris/hello-world-competition")
```

Comment:
What are our thoughts on a never-ending `hello-world` competition hosted by us? Other competition sites do this and it is really good for onboarding newcomers to the feature.

Reply:
Good points. Once we better understand how we will evaluate results, and if there is some possibility of auto-eval in our future, we could return to this idea.

For now, I'll update that artifact ID to a placeholder so users aren't hit with a 404 if they copy/paste it.